analyses/cell-type-dsrct/exploratory_analysis/01-marker-gene-tumor-classification.Rmd

```diff
@@ -0,0 +1,379 @@
+names(sce_list) <- metadata_DSRCT$scpca_library_id
+
+# read in marker genes table 
+marker_genes_df <- readr::read_tsv(marker_genes) |> 
+  # account for genes being from multiple sources
+  dplyr::select(cell_type, ensembl_gene_id, gene_symbol) |> 
+  dplyr::distinct()
+
+marker_genes_df
+
+marker_genes <- marker_genes_df |> 
+  dplyr::filter(cell_type == "tumor") |> 
+  dplyr::pull(ensembl_gene_id)
+```
+
+
+## Analysis content
+
+### Explore marker gene expression 
+
+The first thing we do here is just create a faceted UMAP showing the expression of each marker gene for tumor cells. 
+
+```{r}
+umap_df_list <- lapply(sce_list, function(x) {
+  # get the gene expression counts for all marker genes
+  marker_gene_exp <- logcounts(x[marker_genes, ]) |>
+    as.matrix() |>
+    t() |>
+    as.data.frame() |>
+    tibble::rownames_to_column("barcodes")
+  
+  # pull out the UMAP coordinates and make a data frame to use for plotting
+  umap_df <- x |>
+    scuttle::makePerCellDF(use.dimred = "UMAP") |>
+    # replace UMAP.1 with UMAP1
+    dplyr::rename_with(\(x) stringr::str_replace(x, "^UMAP\\.", "UMAP")) |>
+    # add in marker gene expression to dataframe
+    dplyr::left_join(marker_gene_exp, by = "barcodes") |>
+    # combine all genes into a single column for easy faceting
+    tidyr::pivot_longer(cols = starts_with("ENSG"),
+                        names_to = "ensembl_gene_id",
+                        values_to = "gene_expression") |>
+    # join with marker gene df to get gene symbols for plotting
+    dplyr::left_join(marker_genes_df, by = c("ensembl_gene_id")) |>
+    dplyr::select(barcodes,
+                  UMAP1,
+                  UMAP2,
+                  gene_symbol,
+                  ensembl_gene_id,
+                  gene_expression,
+                  cluster)
+})
+
+names(umap_df_list) <- metadata_DSRCT$scpca_sample_id
+```
+
+
+```{r, fig.width=8}
+for(i in 1:length(umap_df_list)){
+  # faceted umap showing a umap panel for each marker gene
+  p <- ggplot(umap_df_list[[i]], aes(x = UMAP1, y = UMAP2, color = gene_expression)) +
+    geom_point(alpha = 0.8, size = 0.1) +
+    facet_wrap(vars(gene_symbol)) +
+    scale_color_viridis_c() +
+    labs(color = "Log-normalized gene expression") +
+    # remove axis numbers and background grid
+    scale_x_continuous(labels = NULL, breaks = NULL) +
+    scale_y_continuous(labels = NULL, breaks = NULL) +
+    theme(
+      aspect.ratio = 1,
+      legend.position = "bottom",
+      axis.title = element_text(size = 9, color = "black"),
+      strip.text = element_text(size = 8),
+      legend.title = element_text(size = 9),
+      legend.text = element_text(size = 8)
+    ) +
+    guides(colour = guide_colorbar(title.position = "bottom", title.hjust = 0.5)) +
+    ggtitle(names(umap_df_list)[i])
+  
+  tiff(filename = paste0('../plots/marker_expression_', names(umap_df_list)[i], '.tiff'), width = 6, height = 6, units = 'in', res = 150)
+  print(p)
+  dev.off()
```

Comment:
I'd suggest using `ggsave` to allow one-command saving, and `png` should be smaller than tiff (though this may be something to test).

```suggestion
  ggsave(
    filename = paste0('../plots/marker_expression_', names(umap_df_list)[i], '.png'), 
    plot = p,
    width = 6, 
    height = 6, 
    units = 'in', 
    dpi = 150
  )
```

Reply:
Please also include the plot files (and the rendered html if possible) in the submission.

Reply:
Thanks. Will add.